zerver/lib/users.py

```diff
@@ -73,8 +73,11 @@ def check_bot_name_available(realm_id: int, full_name: str) -> None:
 
 def check_short_name(short_name_raw: str) -> str:
     short_name = short_name_raw.strip()
-    if len(short_name) == 0:
-        raise JsonableError(_("Bad name or username"))
+    short_name_pattern_match = re.match(
+        r"^[\w!#$%&'*+/=?^`{|}~-]+(\.[\w!#$%&'*+/=?^`{|}~-]+)*$", short_name, re.I
+    )
+    if len(short_name) == 0 or not short_name_pattern_match:
+        raise JsonableError(_("Invalid input for email"))
```

Comment:
I am not sure about the error handling here. The API parameters here is `short_name` while in webapp the input is labelled as email, I am not sure what would be the correct error message.

Reply:
One possible solution could be to check the validity of email in frontend only before making the request to server like we did before and showing the error accordingly.

Reply:
Wouldn't we need the same validator for the backend as well because via API we can send `short_name` with "@"?

Reply:
I meant we would be validating the email both in backend as well as frontend. And in webapp, we would not be sending the API request to server if the email is incorrect.

Reply:
Okay done.

Reply:
Updated both commits.